src/main/java/org/folio/innreach/batch/contribution/service/ContributionJobRunner.java

```diff
@@ -64,57 +73,125 @@ public class ContributionJobRunner {
+  Integer totalRecords;
+  private static ConcurrentHashMap<String, Integer> recordsProcessed = new ConcurrentHashMap<>();
+
+  public void startInitialContribution(UUID centralServerId, String tenantId, UUID contributionId, UUID iterationJobId, Integer numberOfRecords) {
     var context = ContributionJobContext.builder()
       .contributionId(contributionId)
       .iterationJobId(iterationJobId)
       .centralServerId(centralServerId)
       .tenantId(tenantId)
       .build();
 
-    runInitialContribution(context);
+    totalRecords = numberOfRecords;
+
+    recordsProcessed.put(tenantId, 0);
+
+    InitialContributionJobConsumerContainer tempKafkaConsumer = itemReaderFactory.createInitialContributionConsumerContainer(tenantId);
 
-    return new AsyncResult<>(null);
+    InitialContributionMessageListener initialContributionMessageListener = new InitialContributionMessageListener(new ContributionProcessor(this), context, new Statistics());
+
+    tempKafkaConsumer.tryStartOrCreateConsumer(initialContributionMessageListener);
   }
 
-  public void runInitialContribution(ContributionJobContext context) {
-    log.info("Starting initial contribution job {}", context);
+  public void runInitialContribution(ContributionJobContext context, InstanceIterationEvent event, Statistics stats, String topic) {
+    recordsProcessed.put(context.getTenantId(), recordsProcessed.get(context.getTenantId()) == null ? 1 : recordsProcessed.get(context.getTenantId())+1);
 
     var contributionId = context.getContributionId();
-    try (var kafkaReader = itemReaderFactory.createReader(context.getTenantId())) {
-      kafkaReader.open();
-
-      run(context, (centralServerId, stats) -> {
-        while (!isCanceled(contributionId)) {
-          var event = readEvent(kafkaReader);
-          if (event == null) {
-            return;
-          }
-          log.info("Processing instance iteration event = {}", event);
-
-          var instanceId = event.getInstanceId();
-          var iterationJobId = context.getIterationJobId();
-
-          if (isUnknownEvent(event, iterationJobId)) {
-            log.info("Skipping unknown event, current job is {}", iterationJobId);
-            continue;
-          }
-
-          var instance = loadInstanceWithItems(instanceId);
-          if (instance == null) {
-            continue;
-          }
-
-          if (isEligibleForContribution(centralServerId, instance)) {
-            contributeInstance(centralServerId, instance, stats);
-            contributeInstanceItems(centralServerId, instance, stats);
-          } else if (isContributed(centralServerId, instance)) {
-            deContributeInstance(centralServerId, instance, stats);
-          }
-        }
-      });
+
+    var iterationJobId = context.getIterationJobId();
+
+    if (event == null) {
+      log.info("Exiting kafka message reader for contribution {} and job {}", contributionId, iterationJobId);
+      return;
+    }
```

Comment:
I don't think we want this block any more do we? Or maybe we want to keep the null check (so it doesn't blow up if it gets one), but change the log message to something more accurate to what would happen in this case.